src/widgets/barchart.rs

```diff
@@ -374,6 +412,15 @@ impl<'a> BarChart<'a> {
                     y: bar_y + (self.bar_width >> 1),
                     ..bars_area
                 };
+
+                // label
+                buf.set_line(
```

Comment:
I would check for the label first to avoid calling `set_line`.
also you can avoid cloning the label

```suggestion
                if let Some(label) = &bar.label {
                    buf.set_line(label_x, bar_value_area.top(), label, label_size);
                }
```